src/ufuncs.c

```diff
@@ -43,6 +43,17 @@
   Py_XDECREF(*out);                                      \
   *out = ret
 
+// Fail if inputs output multiple times on the same place in memory. That would
+// lead to segfaults as the same GEOSGeometry would be 'owned' by multiple PyObjects.
+#define CHECK_NO_INPLACE_OUTPUT(N)                                                \
+  if ((steps[N] == 0) && (dimensions[0] > 1)) {                                   \
+    PyErr_Format(PyExc_NotImplementedError,                                       \
```

Comment:
@caspervdw do you remember if it's actually possible to construct an example ufunc call in python that triggers this exception? (I thought to add a test for it, but wasn't able to get one, tried a few things based on the discussion in #156)

Reply:
No I don't think that is an option. But at the same times, I didn't find any guarantees in numpy that this should not happen.

I see this as a forward compatibility. It is a cheap check and it may prevent segfaults when using this pygeos version with future numpy versions.